### ISSUE TITLE
Is it considered bad code quality if two methods are highly "coupled"?

### ISSUE THREAD

**Issue by CSmortal** [nus-cs2103-AY2021S2/forum#244]
So if I have a method `abc` and `xyz`, and `abc` returns a boolean based on a condition that `xyz` ALSO relies on, and for example this is how my code looks like

```
if (abc()) {
      return xyz();
}
```
should `xyz` check for the condition (and also error handle EXACTLY the same as `abc`, if the condition check fails) as well to "decouple" these two methods, or should I instead "couple" these two methods by throwing an AssertionError in the body of `xyz` if `xyz` fails the condition check? 

If the former is preferred, there might be noticeable code duplication as well

**Comment by damithc** [comment]
@CSmortal There is not much value in (and low chance of success in) trying to find general right-or-wrong answers to this kind of questions. It is better to present the actual code and discuss pros-and-cons instead.

**Comment by CSmortal** [comment]
I added in the tp such that deletion of tasks can be done in 1 command, i.e something like `delete 1 2 3`

This is how the `parseCommand` method in the class that my group has renamed from `AddressBookParser` looks like
![image](https://user-images.githubusercontent.com/67994626/111896751-b1382780-8a56-11eb-8ae5-35d6110b7074.png)

This is the `Parser` for the new command `DeleteMultipleCommand`
![image](https://user-images.githubusercontent.com/67994626/111896893-9a460500-8a57-11eb-85d5-316ff531131a.png)

This is the `hasMultipleValidIndex` method that was in the if clause of the very first picture
![image](https://user-images.githubusercontent.com/67994626/111896949-0163b980-8a58-11eb-9687-4662e1df2273.png)

This is the `parseMultipleIndex` method that would be called in the `parse` method of `DeleteMultipleCommandParser`, if `hasMultipleValidIndex()` is true (see the first pic)
![image](https://user-images.githubusercontent.com/67994626/111896992-5b647f00-8a58-11eb-9a53-9d1d6549278f.png)

I wrote the code as such, because i didn't want `parseMultipleIndex` to depend heavily on `hasMultipleValidIndex`. But after looking at it again, I thought if it would be better to throw an AssertionError in both of the 2 condition checks in `parseMultipleIndex`, since it was checked for previously in `hasMultipleValidIndex`. But then I thought it might not be such a good idea since it "couples" these two methods together strongly.

My question is which approach is better for code quality? (Not sure if thats the term to use either)
